presets/workspace/models/metadata.go

```diff
@@ -15,6 +16,9 @@ var (
 	//go:embed supported_models.yaml
 	supportedModelsYAML []byte
 
+	// mutex is used to synchronize access to the SupportedModels map.
+	mutex sync.Mutex
+
 	// SupportedModels is a global map that holds the source of truth
 	// for all supported models and their metadata.
 	SupportedModels map[string]*model.Metadata
```

Comment:
Consider making `SupportedModels` a `sync.Map`. It's designed for this use case.
> (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow,

ref:  https://pkg.go.dev/sync#Map